### DISCUSSION TITLE
Include __typename on auto-generated queries

### DISCUSSION THREAD

**Discussion by mAAdhaTTah** [tinacms/tinacms#3481] [Ideas]
I have a Post type with several templates at the root. When using the built in `postConnection` query, the `__typename` of those posts isn't included in the response, which makes it impossible to switch behaviors based on the different templates. It appears to be included correct when the templates are for a given field (e.g. array of blocks-style setups) but not when it's at the root of the collection.

Can this be added?

ETA: Would be happy to contribute a PR here too if it would be accepted.

**Comment by jmarks-joshua** [comment]
Have exactly the same issue, given that this discussion is over a year old with no comments I'm now feeling disheartened. Did you find a way around this? 

My next approach is just duck typing and casting but this seems suboptimal.

**Reply by jmarks-joshua** [reply]
#4225 There is a PR for this change and has been merged in, my version was out of date
